test/extended/deployments/deployments.go

```diff
@@ -326,14 +326,15 @@ var _ = g.Describe("[Feature:DeploymentConfig] deploymentconfigs", func() {
 
 			out, err := oc.Run("logs").Args("-f", "dc/deployment-test").Output()
 			o.Expect(err).NotTo(o.HaveOccurred())
+
+			g.By("verifying the deployment is marked complete and scaled to zero")
```

Comment:
why this move? AFAICS it doesn't change the order since the result of `oc logs -f` is stored in `out` already

Reply:
my understanding is that w/o verifing completion, logs is racy and we'll loose logs (I may be wrong)

Reply:
But you have already got the data from API in the line above so moving comparing to internal memory buffers changes nothing at all.

There is a race not related to your change. Actually if you were to move it a bit above (probably what you wanted) before `oc.Run("logs").Args("-f", "dc/deployment-test").Output()` that would have made it even worse I think. The race is between the controller deleting the deployer pod (after finishing) and calling `oc logs`. You want to get the logs asap, with -f option to follow, or the controller can manage to delete the pods and logs will be gone.